### PR TITLE
adding utility function to handle namespace changes

### LINKED ISSUE #217: Parse namespaces out of HTML
As a short term fix for the namespace issue in the editor, we will parse the namespaces from the HTML on save.

### PR DESCRIPTION
**Pull request by rich-hart**
Fixes #217

### CODE REVIEW

File: cnxauthoring/tests/test_views.py [hunk 1]
```diff
@@ -72,6 +72,54 @@ def storage_cls(self):
         from ..storage.main import BaseStorage
         return BaseStorage
 
+    def test_update_content_state_with_parsing(self):
```
Comment (rich-hart) [inline cnxauthoring/tests/test_views.py]: I still need to write a utility test. Also, I might change the names of some variables like this one.

File: cnxauthoring/models.py [hunk 2]
```diff
@@ -194,6 +194,8 @@ def __init__(self, title, acls=None,
 
     def update(self, **kwargs):
         super(Document, self).update(**kwargs)
+        self.metadata['content'] = utils.manage_namespace(
```
Comment (rich-hart) [inline cnxauthoring/models.py]: change name from ```utils.manage_namespace --> utils.update_namespace```

File: cnxauthoring/tests/data/m.json [hunk 3]
```diff
@@ -0,0 +1,2 @@
...ML\">+</mo>\n  <msup xmlns=\"http://www.w3.org/1998/Math/MathML\">\n    <mi xmlns=\"http://www.w3.org/1998/Math/MathML\">x</mi>\n    <mn xmlns=\"http://www.w3.org/1998/Math/MathML\">3</mn>\n  </msup>\n  <mo xmlns=\"http://www.w3.org/1998/Math/MathML\">=</mo>\n  <msup xmlns=\"http://www.w3.org/1998/Math/MathML\">\n    <mi xmlns=\"http://www.w3.org/1998/Math/MathML\">y</mi>\n    <mn xmlns=\"http://www.w3.org/1998/Math/MathML\">2</mn>\n  </msup>\n</mrow><annotation xmlns=\"http://www.w3.org/1998/Math/MathML\" encoding=\"math/tex\">\\sqrt{36} + x^3 = y^2</annotation></semantics></math>\n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  \n  <br/></p>", "state": "Draft", "subjects": [], "copyrightHolders": [{"website": null, "hasAccepted": true, "surname": "Reedstrom", "firstname": "Ross", "title": "Dr", "emails": ["ross.reedstrom@rice.edu", "ross@cnx.org"], "id": "reedstrm", "requester": "reedstrm", "assignmentDate": "2015-02-10T14:57:51.543655-06:00", "fullname": "Ross Reedstrom", "type": "cnx-id", "email": ""}], "submitter": {"surname": "Reedstrom", "fullname": "Ross J. Reedstrom", "type": "cnx-id", "id": "reedstrm", "firstname": "Ross"}, "illustrators": [], "translators": []}
```
Comment (rich-hart) [inline cnxauthoring/tests/data/m.json]: change ``` m.json --> namespace_test.json ```

File: cnxauthoring/utils.py [hunk 4]
```diff
@@ -43,6 +43,31 @@
...
+              'xmlns:data="http://dev.w3.org/html5/spec/#custom" '
+              'xmlns:epub="http://www.idpf.org/2007/ops" '
+              'xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" '
+              'xmlns:dc="http://purl.org/dc/elements/1.1/" '
+              'xmlns:lrmi="http://lrmi.net/the-specification">'
+              '{}'
+              '</body>')
+
+
+def manage_namespace(content):
```
Comment (rich-hart) [inline cnxauthoring/utils.py]: name change: ``` manage_namespace --> update_namespace ```

**Comment by pumazi** [comment]
I think this fix should have been done in the content setter within the cnx-epub model.

File: cnxauthoring/utils.py [hunk 5]
```diff
@@ -43,6 +43,31 @@
...
+    try:
+        e = etree.fromstring(content)
+    except etree.XMLSyntaxError:
+        ##################################################
+        # trying to catch error by wrapping content in a #
+        # namespaced body tag                            #
+        ##################################################
```
Comment (reedstrm) [inline cnxauthoring/utils.py]: A comment like:
Try and  handle two cases: no root tag (wrap in a <body>) and duplicate xmlns= (upstream error in mathml - not sure if it's mathjax or mathjax/aloha plugin) Note the "big hammer" of recover for that second case.
BBB remove recover once upstream dup xmlns is solved

**Pull request merged**